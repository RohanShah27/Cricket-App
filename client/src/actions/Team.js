import { GET_TOURNAMENT } from "./Types";
import axios from "axios";
const url = "http://localhost:5000/api/team";

export const getTournament = tournament => dispatch => {
  console.log(tournament);
  return axios.post(url + "/tournament", tournament).then(res => {
    dispatch({
      type: GET_TOURNAMENT,
      payload: res.data.data
    });
    console.log(res.data.data);
  });
};
