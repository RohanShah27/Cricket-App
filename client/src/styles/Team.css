@import url(https://fonts.googleapis.com/css?family=Work+Sans);

@media only screen and (max-width: 425px) {
  .team-search-box > input {
    width: 200px;
    margin-top: 2%;
  }

  .team-search-button {
    margin-top: 2%;
    margin-right: 20%;
  }

  .pc-playerTab ul li label {
    width: auto;
    font-size: 10px;
    padding: 10px !important;
  }

  .pc-playerTab ul li.playerTab4 > label {
    margin-right: 40px;
  }

  .pc-playerTab nav {
    margin-left: 4%;
  }
}

*/ *::-webkit-scrollbar {
  display: none;
}
.team-body {
  margin: auto;
  padding: 0;
  background-color: white;
  font-family: "Work Sans";
  width: 100%;
  height: 100%;
}

.inputforbuttons,
.pc-playerTab section > div {
  display: none;
}

#playerTab1:checked ~ section .playerTab1,
#playerTab2:checked ~ section .playerTab1,
#playerTab3:checked ~ section .playerTab1,
#playerTab4:checked ~ section .playerTab1 {
  display: block;
}

*,
*:after,
*:before {
  box-sizing: border-box;
}

.pc-playerTab {
  width: 100%;
  margin: 0 auto;
  margin-top: 5%;
}

.pc-playerTab ul {
  list-style-type: none;
  margin-left: 2%;
  padding: 0;
}

.pc-playerTab ul li label {
  float: left;
  padding: 15px 25px;
  border: 1px solid #f39c12;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
}

.pc-playerTab ul li label:hover {
  background-color: #f39c12;
  color: white;
  border: 1px solid white;
}

/* .pc-playerTab ul li:not(:last-child) label {
  border-right-width: 0;
} */

.pc-playerTab section {
  clear: both;
}

.pc-playerTab section .playerTab1 {
  padding: 20px;
  width: 100%;
  height: 100vh;
  overflow: auto;
  border-left-color: white;
  border-right-color: white;
  border-bottom-color: white;
}
li {
  user-select: none;
}

.pc-playerTab section div h2 {
  margin: 0;
}

/* when on that playerTab */
#playerTab1:checked ~ nav .playerTab1 label,
#playerTab2:checked ~ nav .playerTab2 label,
#playerTab3:checked ~ nav .playerTab3 label,
#playerTab4:checked ~ nav .playerTab4 label {
  position: relative;
  background-color: #f39c12;
  color: white;
  border: 1px solid white;
}

#playerTab1:checked ~ nav .playerTab1 label:after,
#playerTab2:checked ~ nav .playerTab2 label:after,
#playerTab3:checked ~ nav .playerTab3 label:after,
#playerTab4:checked ~ nav .playerTab4 label:after {
  display: block;
  position: absolute;
  height: 2px;
  width: 100%;
  left: 0;
  bottom: -1px;
}

.team-teamTestimonials {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  grid-gap: 10px;
}

.team-teamTestimonials .teamcomponent-card {
  position: relative;
  width: 300px;
  height: 220px;
  margin: 0 auto;
  text-align: center;
  border: 1px solid #c2c2c2;
  border-radius: 8px;
  overflow: hidden;
  cursor: pointer;
}

.team-teamTestimonials .teamcomponent-card:hover {
  box-shadow: 0px 4px 6px 0px #c2c2c2;
}

.team-teamTestimonials .teamcomponent-card .team-content {
  z-index: 2;
}

.team-teamTestimonials .teamcomponent-card .team-content p {
  font-size: 18px;
  line-height: 24px;
}

.playerViewDetails {
  border: 1px solid #f39c12;
  border-radius: 8px;
  color: #f39c12;
  background-color: #ffffff;
  margin-top: 5%;
  width: 100px;
  height: 40px;
  outline: none;
  cursor: pointer;
}

.playerViewDetails:hover {
  color: white;
  background: #f39c12;
}

.iplLogo {
  margin-top: 10px;
  border-radius: 50%;
  width: 30%;
  height: 100px;
}

.internationalLogo {
  margin-top: 20px;
  width: 35%;
  height: 100px;
}

.team-search-box {
  margin-right: 2%;
}

.team-search-input {
  color: #f39c12;
  float: right;
  width: 300px;
  height: 40px;
  padding: 12px;
  border: 1px solid #c2c2c2;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.team-search-input:focus {
  outline: none;
}

.team-search-button {
  height: 40px;
  width: 30px;
  float: right;
  cursor: pointer;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
  color: white;
  border: none;
  background: #f39c12;
}

/* .team-search-box; */
