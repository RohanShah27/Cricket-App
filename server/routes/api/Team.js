const express = require("express");
const router = express.Router();
const pg = require("pg-promise")();
const db = pg("postgres://postgres:postgres@localhost/crickstrait_capstone");

// router.post("/tournament", async (req, res) => {
//   const tournament = req.body.tournament;
//   console.log(tournament);
//   const result = await db.any(
//     "select * from teams where tournament ='" + tournament + "';"
//   );
//   res.status(200).json({
//     status: 200,
//     data: result,
//     message: "Retrieved all tournaments  successfully"
//   });
// });

router.post("/tournament", async (req, res) => {
  const tournament = req.body.tournament;
  console.log(tournament);
  const result = await db.any(
    `with ss as(select match.innings_one_team, match.innings_two_team, match.competition, team.team_name, team.team_id ,team.team_image,team.team_background from match 
      inner join team on match.innings_one_team = team.team_id), 
      t2 as (select match.innings_two_team, match.competition, team.team_name, team.team_id,team.team_image,team.team_background  from match 
      inner join team on match.innings_two_team = team.team_id) select distinct(team_name), team_id,team_image,team_background  from t2 where competition='${tournament}'`
  );
  res.status(200).json({
    status: 200,
    data: result,
    message: "Retrieved all tournaments  successfully"
  });
});

router.post("/playerstatsforteams/:team_id", async (req, res) => {
  const match_type = req.body.match_type;
  const team_id = req.params.team_id;
  const result = await db.any(
    `Select distinct player_name, player_stats_value from teamplayer_stats where player_stats_name = 'total_runs' and team_id= ${team_id} and match_type = '${match_type}' order by player_stats_value desc fetch first 5 rows only`
  );
  res.status(200).json({
    status: 200,
    data: result,
    message: "Retrieved all data successfully"
  });
});

router.post("/playerstatsforteamsbowler/:team_id", async (req, res) => {
  const match_type = req.body.match_type;
  const team_id = req.params.team_id;
  const result = await db.any(
    `Select distinct player_name, player_stats_value from teamplayer_stats where player_stats_name = 'total_wickets' and team_id= ${team_id} and match_type = '${match_type}' order by player_stats_value desc fetch first 5 rows only`
  );
  res.status(200).json({
    status: 200,
    data: result,
    message: "Retrieved all data successfully"
  });
});

router.post("/teambyid", async (req, res) => {
  const team_id = req.body.team_id;
  console.log(team_id);
  const result = await db.any(
    "select * from team where team_id = '" + team_id + "';"
  );
  res.status(200).json({
    status: 200,
    data: result,
    message: "Retrieved team successfully"
  });
});

router.get("/matchtype/:team_id", async (req, res) => {
  try {
    const team_id = req.params.team_id;
    console.log(team_id);
    const result = await db.any(
      `Select 
      (Select count(match_id) from match where (innings_one_team=${team_id} or innings_two_team=${team_id}) and match_type='Test') AS Test_played,
     (Select count(match_id) from match where (innings_one_team=${team_id} or innings_two_team=${team_id}) and match_type='ODI') AS ODI_played,
      (Select count(match_id) from match where (innings_one_team=${team_id} or innings_two_team=${team_id}) and match_type='T20') AS T20_played,
      (Select count(winner) from match where winner = ${team_id} and match_type='Test') AS Test_win,
    (Select count(winner) from match where winner = ${team_id} and match_type='ODI') AS Odi_win,
    (Select count(winner) from match where winner = ${team_id} and match_type='T20') AS T20_win,
    (Select count(winner) from match where (innings_one_team=${team_id} or innings_two_team=${team_id}) and not winner = ${team_id} and not winner=0 and match_type='Test') AS Test_loss,
 (Select count(winner) from match where (innings_one_team=${team_id} or innings_two_team=${team_id}) and not winner = ${team_id} and not winner=0 and match_type='ODI') AS ODI_loss,
 (Select count(winner) from match where (innings_one_team=${team_id} or innings_two_team=${team_id}) and not winner = ${team_id} and not winner=0 and match_type='T20') AS T20_loss,
 
      (Select count(winner) from match where  (innings_one_team=${team_id} or innings_two_team=${team_id}) and winner=0 and match_type='Test') AS Test_draw,
 (Select count(winner) from match where  (innings_one_team=${team_id} or innings_two_team=${team_id}) and winner=0 and match_type='ODI') AS ODI_draw,
 (Select count(winner) from match where  (innings_one_team=${team_id} or innings_two_team=${team_id}) and winner=0 and match_type='T20') AS T20_draw
 


      
    
      `
    );
    res.status(200).json({
      status: 200,
      data: result,
      message: "Retrieved search info successfully"
    });
  } catch (err) {
    res.status(400);
  }
});

router.get("/playerstats", async (req, res) => {
  const result = await db.any(
    "Select player_stats.match_type, player_stats.player_stats_name,player_stats.player_stats_value from team where team_id = '" +
      team_id +
      "';"
  );
  res.status(200).json({
    status: 200,
    data: result,
    message: "Retrieved team successfully"
  });
});

router.post("/teamsearch", async (req, res) => {
  try {
    let team_name = req.body.team_name;
    let tournament = req.body.tournament;
    console.log(team_name);
    const query = `select * from team where team_name ilike '${team_name}%'`;
    console.log("query", query);
    const result = await db.any(
      query

      // "select * from teams where team_name = '" +
      //   team_name +
      //   "'and tournament='" +
      //   tournament +
      //   "';"
    );

    if (result.length == 0) {
      res.status(404).json({
        statusCode: 404,
        message: "Cannot find team with the specified name",
        data: null
      });
    }
    res.status(200).json({
      status: 200,
      data: result,
      message: "Retrieved team successfully"
    });
  } catch (err) {
    res.status(400).json({
      status: 400,
      message: "Server Error",
      data: null
    });
  }
});

module.exports = router;
