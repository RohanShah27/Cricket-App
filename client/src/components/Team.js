import React, { Component } from "react";
import { connect } from "react-redux";
import "../styles/Team.css";
import InfiniteScroll from "react-infinite-scroll-component";
import { Link } from "react-router-dom";
import ReactCountryFlag from "react-country-flag";

import {
  getTournament,
  searchTeamForViewTeamPage
} from "../actions/teamActions";

export class Team extends Component {
  constructor(props) {
    super(props);
    this.sendTeam = this.sendTeam.bind(this);
  }

  state = {
    tournament: "",
    team_name: "",
    items: 8,
    loadingState: false,
    height: 800,
    pageTeams: []
  };

  componentDidMount() {
    let team = { tournament: "others" };
    this.props.getTournament(team);
  }

  componentWillReceiveProps(nextProps) {
    nextProps.tournamentTeam.length > 0
      ? this.displayTeams(nextProps.tournamentTeam)
      : console.log(0, " teams");
  }
  OnChange = event => {
    this.setState({ [event.target.name]: event.target.value });

    let team_name = {
      team_name: this.state.team_name
    };
    this.props.searchTeamForViewTeamPage(team_name);
  };

  getTeam = () => {
    let team = {
      team_name: this.state.team_name,
      tournament: this.state.tournament
    };
    console.log(team);
    this.props.searchTeamForViewTeamPage(team);
  };

  sendTeam(team) {
    this.setState({
      tournament: team
    });
    let tournament = {
      tournament: this.state.tournament
    };
    this.props.getTournament(tournament);
  }

  displayTeams = teams => {
    console.log(teams);
    const { items } = this.state;

    if (teams.length === 0) return;
    let pageTeams = [];
    for (let i = 0; i < items; i++) {
      pageTeams.push(teams[i]);
    }
    this.setState({ pageTeams });
    console.log("PageTeams array", this.state.pageTeams.length);
  };

  loadMoreItems = () => {
    setTimeout(() => {
      this.setState({
        items: this.state.items + 8
      });

      console.log("Teams", this.props.tournamentTeam);
      this.displayTeams(this.props.tournamentTeam);
    }, 1000);
  };
  render() {
    return (
      <div className="team-body">
        <div className="pc-playerTab">
          <input
            className="inputforbuttons"
            defaultChecked="defaultChecked"
            id="playerTab1"
            type="radio"
            name="pct"
          />
          <input
            className="inputforbuttons"
            id="playerTab2"
            type="radio"
            name="pct"
          />
          <input
            className="inputforbuttons"
            id="playerTab3"
            type="radio"
            name="pct"
          />
          <input
            className="inputforbuttons"
            id="playerTab4"
            type="radio"
            name="pct"
          />
          <nav>
            <ul>
              <li className="playerTab1">
                <label
                  id="othersTab"
                  htmlFor="playerTab1"
                  onClick={() => {
                    this.sendTeam("others");
                  }}
                >
                  International
                </label>
              </li>
              <li className="playerTab2">
                <label
                  id="iplTab"
                  htmlFor="playerTab2"
                  onClick={() => {
                    this.sendTeam("IPL");
                  }}
                >
                  IPL
                </label>
              </li>
              <li className="playerTab3">
                <label
                  id="BBLTab"
                  htmlFor="playerTab3"
                  onClick={() => {
                    this.sendTeam("Big Bash League");
                  }}
                >
                  Big Bash League
                </label>
              </li>
              <li className="playerTab4">
                <label
                  id="PSLTab"
                  htmlFor="playerTab4"
                  onClick={() => {
                    this.sendTeam("Pakistan Super League");
                  }}
                >
                  Pakistan Super League
                </label>
              </li>
            </ul>
          </nav>
          <div className="team-search-box">
            <button
              className="team-search-button"
              id="searchTeamButton"
              onClick={() => this.getTeam}
            >
              <i class="fa fa-search"></i>
            </button>
            <input
              id="searchTeamTextBox"
              className="team-search-input"
              type="text"
              placeholder="Enter Team Name"
              name="team_name"
              onChange={this.OnChange}
            />
            {this.state.team_name == "" ? null : (
              <div className="team-search-result">
                <span className="search_result_data">
                  <p>
                    {this.props.teamsearch
                      ? this.props.teamsearch.map(teams => (
                          <Link
                            style={{
                              textDecoration: "none"
                            }}
                            to={{
                              pathname: "/viewteam/" + teams.team_id,
                              state: { teams }
                            }}
                          >
                            {" "}
                            <p
                              style={{
                                fontSize: "18px",
                                paddingTop: "10px ",
                                display: "flex",
                                justifyContent: "center"
                              }}
                            >
                              {teams.team_name}
                            </p>
                          </Link>
                        ))
                      : null}
                  </p>
                </span>
              </div>
            )}
          </div>
          <section>
            <div className="playerTab1">
              <InfiniteScroll
                dataLength={this.state.pageTeams.length} //This is important field to render the next data
                next={this.loadMoreItems}
                hasMore={true}
                height={600}
                loader={
                  <div className="loader-container">
                    <div className="user-loader"></div>
                  </div>
                }
                endMessage={
                  <p style={{ textAlign: "center" }}>
                    <b>Yay! You have seen all Teams</b>
                  </p>
                }
              >
                <div className="team-teamTestimonials">
                  {this.state.pageTeams.map((teams, index) => {
                    return (
                      <div>
                        {teams ? (
                          <div className="teamcomponent-card">
                            <div className="team-content">
                              {/* <p>Started</p> */}
                              <ReactCountryFlag
                                styleProps={{
                                  width: "70px",
                                  height: "70px"
                                }}
                                code={
                                  teams.team_image ? teams.team_image : "ao"
                                }
                                svg
                                className="internationalLogo"
                              />{" "}
                              <p id={"teamName" + index}>{teams.team_name}</p>
                              {/* <p>India</p> */}
                              <div className="team-details">
                                <p>
                                  <button
                                    id={"teamDetailsButton" + index}
                                    className="playerViewDetails"
                                    onClick={() => {
                                      this.props.history.push(
                                        "/viewteam/" + teams.team_id,
                                        { teams }
                                      );
                                    }}
                                  >
                                    {" "}
                                    View Details
                                  </button>
                                </p>
                              </div>
                            </div>
                            {/* </div> */}
                          </div>
                        ) : (
                          <div></div>
                        )}
                      </div>
                    );
                  })}
                </div>
              </InfiniteScroll>
            </div>
          </section>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  tournamentTeam: state.teamReducer.tournamentTeam,
  teamsearch: state.teamReducer.teamsearch
});

export default connect(
  mapStateToProps,
  { getTournament, searchTeamForViewTeamPage }
)(Team);
