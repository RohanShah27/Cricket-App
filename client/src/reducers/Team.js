import { GET_TEAM, GET_TOURNAMENT } from "../actions/Types";

const initialState = {
  tournamentTeam: []
};
export default function(state = initialState, action) {
  switch (action.type) {
    case GET_TEAM:
      return {
        ...state,
        team: action.payload
      };
    case GET_TOURNAMENT:
      return {
        ...state,
        tournamentTeam: action.payload
      };
    default:
      return state;
  }
}
