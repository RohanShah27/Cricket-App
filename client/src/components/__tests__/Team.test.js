import React from "react";
import { mount } from "enzyme";
import { Team } from "../Team";

const history = { push: jest.fn() };
let teams = [];
let state = {
  height: 800,
  items: 8,
  loadingState: false,
  pageTeams: [
    { team_id: 0, team_name: "test name" },
    { team_id: 1, team_name: "test name" },
    { team_id: 2, team_name: "test name" },
    { team_id: 3, team_name: "test name" },
    { team_id: 4, team_name: "test name" },
    { team_id: 5, team_name: "test name" },
    { team_id: 6, team_name: "test name" },
    { team_id: 7, team_name: "test name" }
  ],
  team_name: "India",
  tournament: "Pakistan Super League"
};
for (var i = 0; i < 21; i++) {
  teams.push({
    team_id: i,
    team_name: "test name"
  });
}
let tournamentTeam = [];
for (var i = 0; i < 21; i++) {
  tournamentTeam.push({
    team_id: i,
    team_name: "test name"
  });
}
const getTeam = jest.fn();
const getTournament = jest.fn();
const searchTeamForViewTeamPage = jest.fn();
const sendTeam = jest.spyOn(Team.prototype, "sendTeam");
const wrapper = mount(
  <Team
    getTeam={getTeam}
    teams={teams}
    history={history}
    getTournament={getTournament}
    tournamentTeam={tournamentTeam}
    searchTeamForViewTeamPage={searchTeamForViewTeamPage}
  />
);

describe("Test Team Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should check for component did mount as a function and to call function getTournament", () => {
    const componentDidMount = jest.spyOn(Team.prototype, "componentDidMount");
    wrapper.instance().componentDidMount();
    expect(componentDidMount).toHaveBeenCalled();
    let team = {
      tournament: "others"
    };
    expect(getTournament).toHaveBeenCalledWith(team);
  });
  it("should check onChange as a function", () => {
    let event = {
      target: {
        name: "team_name",
        value: "India"
      }
    };
    wrapper.instance().OnChange(event);
    expect(wrapper.state().team_name).toBe(event.target.value);
    expect(searchTeamForViewTeamPage).toBeCalledWith({
      team_name: event.target.value
    });
  });
  it("should check for search bar on teams page", () => {
    expect(wrapper.find("#searchTeamTextBox").length).toBe(1);
    expect(wrapper.find("#searchTeamTextBox").props().placeholder).toBe(
      "Enter Team Name"
    );
  });
  it("should check for search bar button and its function", () => {
    expect(wrapper.find("#searchTeamButton").length).toBe(1);
    expect(wrapper.find("#searchTeamButton").simulate("click"));
    const getTeam = jest.spyOn(wrapper.instance(), "getTeam");
    expect(getTeam).toBeCalledTimes(0);
  });
  it("should check if Infinite Scroll component is present or not", () => {
    expect(wrapper.find("InfiniteScroll").exists()).toBeTruthy();
  });
  it("should check for othersTab", () => {
    expect(wrapper.find("#othersTab").simulate("click"));
    expect(sendTeam).toHaveBeenCalledWith("others");
  });
  it("should check for IPLTab", () => {
    expect(wrapper.find("#iplTab").simulate("click"));
    expect(sendTeam).toHaveBeenCalledWith("IPL");
  });
  it("should check for BBLTab", () => {
    expect(wrapper.find("#BBLTab").simulate("click"));
    expect(sendTeam).toHaveBeenCalledWith("Big Bash League");
  });
  it("should check for PSLTab", () => {
    expect(wrapper.find("#PSLTab").simulate("click"));
    expect(sendTeam).toHaveBeenCalledWith("Pakistan Super League");
  });
  it("shouuld check for Component will recieve props function", () => {
    let componentWillReceiveProps = jest.spyOn(
      Team.prototype,
      "componentWillReceiveProps"
    );
    let nextProps = {
      tournamentTeam: tournamentTeam
    };
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(componentWillReceiveProps).toHaveBeenCalledWith(nextProps);
    let displayTeams = jest.spyOn(wrapper.instance(), "displayTeams");
    expect(displayTeams).toBeCalledTimes(0);
  });
  it("should expect the state to be latest data", () => {
    expect(wrapper.state()).toStrictEqual(state);
  });
  it("shouuld check for mapping of teams data", () => {
    expect(wrapper.find("#teamName0")).toBeTruthy();
    expect(wrapper.find("#teamName0").props().children).toBe("test name");
  });
  it("Check history push as a function ", () => {
    // expect(wrapper.state().pageTeams).toBe("");
    wrapper.find("#teamDetailsButton0").simulate("click");
    expect(history.push).toBeCalledWith("/viewteam/" + 1, {
      team_id: 1,
      team_name: "test name"
    });
  });
});
