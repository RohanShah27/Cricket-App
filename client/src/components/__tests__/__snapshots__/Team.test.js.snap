// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Team Component should render the component 1`] = `
<Team
  getTeam={[MockFunction]}
  getTournament={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "tournament": "others",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  history={
    Object {
      "push": [MockFunction],
    }
  }
  searchTeamForViewTeamPage={[MockFunction]}
  teams={
    Array [
      Object {
        "team_id": 0,
        "team_name": "test name",
      },
      Object {
        "team_id": 1,
        "team_name": "test name",
      },
      Object {
        "team_id": 2,
        "team_name": "test name",
      },
      Object {
        "team_id": 3,
        "team_name": "test name",
      },
      Object {
        "team_id": 4,
        "team_name": "test name",
      },
      Object {
        "team_id": 5,
        "team_name": "test name",
      },
      Object {
        "team_id": 6,
        "team_name": "test name",
      },
      Object {
        "team_id": 7,
        "team_name": "test name",
      },
      Object {
        "team_id": 8,
        "team_name": "test name",
      },
      Object {
        "team_id": 9,
        "team_name": "test name",
      },
      Object {
        "team_id": 10,
        "team_name": "test name",
      },
      Object {
        "team_id": 11,
        "team_name": "test name",
      },
      Object {
        "team_id": 12,
        "team_name": "test name",
      },
      Object {
        "team_id": 13,
        "team_name": "test name",
      },
      Object {
        "team_id": 14,
        "team_name": "test name",
      },
      Object {
        "team_id": 15,
        "team_name": "test name",
      },
      Object {
        "team_id": 16,
        "team_name": "test name",
      },
      Object {
        "team_id": 17,
        "team_name": "test name",
      },
      Object {
        "team_id": 18,
        "team_name": "test name",
      },
      Object {
        "team_id": 19,
        "team_name": "test name",
      },
      Object {
        "team_id": 20,
        "team_name": "test name",
      },
    ]
  }
  tournamentTeam={
    Array [
      Object {
        "team_id": 0,
        "team_name": "test name",
      },
      Object {
        "team_id": 1,
        "team_name": "test name",
      },
      Object {
        "team_id": 2,
        "team_name": "test name",
      },
      Object {
        "team_id": 3,
        "team_name": "test name",
      },
      Object {
        "team_id": 4,
        "team_name": "test name",
      },
      Object {
        "team_id": 5,
        "team_name": "test name",
      },
      Object {
        "team_id": 6,
        "team_name": "test name",
      },
      Object {
        "team_id": 7,
        "team_name": "test name",
      },
      Object {
        "team_id": 8,
        "team_name": "test name",
      },
      Object {
        "team_id": 9,
        "team_name": "test name",
      },
      Object {
        "team_id": 10,
        "team_name": "test name",
      },
      Object {
        "team_id": 11,
        "team_name": "test name",
      },
      Object {
        "team_id": 12,
        "team_name": "test name",
      },
      Object {
        "team_id": 13,
        "team_name": "test name",
      },
      Object {
        "team_id": 14,
        "team_name": "test name",
      },
      Object {
        "team_id": 15,
        "team_name": "test name",
      },
      Object {
        "team_id": 16,
        "team_name": "test name",
      },
      Object {
        "team_id": 17,
        "team_name": "test name",
      },
      Object {
        "team_id": 18,
        "team_name": "test name",
      },
      Object {
        "team_id": 19,
        "team_name": "test name",
      },
      Object {
        "team_id": 20,
        "team_name": "test name",
      },
    ]
  }
>
  <div
    className="team-body"
  >
    <div
      className="pc-playerTab"
    >
      <input
        className="inputforbuttons"
        defaultChecked="defaultChecked"
        id="playerTab1"
        name="pct"
        type="radio"
      />
      <input
        className="inputforbuttons"
        id="playerTab2"
        name="pct"
        type="radio"
      />
      <input
        className="inputforbuttons"
        id="playerTab3"
        name="pct"
        type="radio"
      />
      <input
        className="inputforbuttons"
        id="playerTab4"
        name="pct"
        type="radio"
      />
      <nav>
        <ul>
          <li
            className="playerTab1"
          >
            <label
              htmlFor="playerTab1"
              id="othersTab"
              onClick={[Function]}
            >
              International
            </label>
          </li>
          <li
            className="playerTab2"
          >
            <label
              htmlFor="playerTab2"
              id="iplTab"
              onClick={[Function]}
            >
              IPL
            </label>
          </li>
          <li
            className="playerTab3"
          >
            <label
              htmlFor="playerTab3"
              id="BBLTab"
              onClick={[Function]}
            >
              Big Bash League
            </label>
          </li>
          <li
            className="playerTab4"
          >
            <label
              htmlFor="playerTab4"
              id="PSLTab"
              onClick={[Function]}
            >
              Pakistan Super League
            </label>
          </li>
        </ul>
      </nav>
      <div
        className="team-search-box"
      >
        <button
          className="team-search-button"
          id="searchTeamButton"
          onClick={[Function]}
        >
          <i
            class="fa fa-search"
          />
        </button>
        <input
          className="team-search-input"
          id="searchTeamTextBox"
          name="team_name"
          onChange={[Function]}
          placeholder="Enter Team Name"
          type="text"
        />
      </div>
      <section>
        <div
          className="playerTab1"
        >
          <InfiniteScroll
            dataLength={0}
            disableBrowserPullToRefresh={true}
            endMessage={
              <p
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <b>
                  Yay! You have seen all Teams
                </b>
              </p>
            }
            hasMore={true}
            height={600}
            loader={
              <div
                className="loader-container"
              >
                <div
                  className="user-loader"
                />
              </div>
            }
            next={[Function]}
            pullDownToRefreshContent={
              <h3>
                Pull down to refresh
              </h3>
            }
            pullDownToRefreshThreshold={100}
            releaseToRefreshContent={
              <h3>
                Release to refresh
              </h3>
            }
          >
            <div
              style={Object {}}
            >
              <div
                className="infinite-scroll-component "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": 600,
                    "overflow": "auto",
                  }
                }
              >
                <div
                  className="team-teamTestimonials"
                />
                <div
                  className="loader-container"
                >
                  <div
                    className="user-loader"
                  />
                </div>
              </div>
            </div>
          </InfiniteScroll>
        </div>
      </section>
    </div>
  </div>
</Team>
`;
